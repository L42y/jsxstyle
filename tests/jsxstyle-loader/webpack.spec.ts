import path = require('path');
import webpack = require('webpack');
import MemoryFS = require('webpack/lib/MemoryOutputFileSystem');
import webpackConfig = require('./webpack/webpack.config');

// one minute
jest.setTimeout(60000);

// TODO: evaluate webpack bundle
it('builds without issue', () => {
  const compiler = webpack(webpackConfig);
  const fs = new MemoryFS();
  compiler.outputFileSystem = fs;

  expect.assertions(2);
  return new Promise((resolve, reject) => {
    compiler.run((err: Error & { details: any }, stats) => {
      if (err) {
        console.error(err.stack || err);
        if (err.details) {
          console.error(err.details);
        }
        return reject(err);
      }

      const info = stats.toJson();
      if (stats.hasErrors()) {
        return reject(info.errors);
      }
      if (stats.hasWarnings()) {
        console.warn(info.warnings);
      }

      const redCSS = fs.readFileSync(
        path.join(webpackConfig.output.path, 'bundle-red.css'),
        'utf8'
      );
      const blueCSS = fs.readFileSync(
        path.join(webpackConfig.output.path, 'bundle-blue.css'),
        'utf8'
      );

      expect(redCSS)
        .toEqual(`/* ./jsxstyle-loader/webpack/test-app/RedApp.js:8 (Inline) */
._1ioutjs {
  color: red;
  display: inline;
}

/* ./jsxstyle-loader/webpack/test-app/Shared.js:8 (Block) */
._1qb53c2 {
  display: block;
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  font-size: 18px;
  line-height: 22px;
}

`);
      expect(blueCSS)
        .toEqual(`/* ./jsxstyle-loader/webpack/test-app/Shared.js:8 (Block) */
._1qb53c2 {
  display: block;
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  font-size: 18px;
  line-height: 22px;
}

/* ./jsxstyle-loader/webpack/test-app/BlueApp.js:8 (Inline) */
._1qr3dx1 {
  color: blue;
  display: inline;
}

`);

      resolve();
    });
  });
});
