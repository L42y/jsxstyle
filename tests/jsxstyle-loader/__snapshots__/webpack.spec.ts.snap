// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds without issue 1`] = `
"/* ./jsxstyle-loader/webpack/test-app/Shared.js:8 (Block) */
._1qb53c2 {
  display: block;
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  font-size: 18px;
  line-height: 22px;
}

/* ./jsxstyle-loader/webpack/test-app/RedApp.js:8 (Inline) */
._1ioutjs {
  color: red;
  display: inline;
}

"
`;

exports[`builds without issue 2`] = `
"/* ./jsxstyle-loader/webpack/test-app/BlueApp.js:8 (Inline) */
._1qr3dx1 {
  color: blue;
  display: inline;
}

/* ./jsxstyle-loader/webpack/test-app/Shared.js:8 (Block) */
._1qb53c2 {
  display: block;
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  font-size: 18px;
  line-height: 22px;
}

"
`;
